packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -98,18 +97,13 @@ export class ValidatorApi implements IValidatorApi {
   public async produceAttestation(validatorPubKey: BLSPubkey, index: CommitteeIndex, slot: Slot): Promise<Attestation> {
     try {
       await this.checkSyncStatus();
-      const [headBlockRoot, {state: headState, epochCtx}] = await Promise.all([
-        this.chain.forkChoice.getHeadRoot(),
-        this.chain.getHeadStateContext(),
-      ]);
+      const headRoot = this.chain.forkChoice.getHeadRoot();
+      const {state, epochCtx} = await this.chain.regen.getBlockSlotState(headRoot, slot);
```

Comment:
I'm pretty sure we only need state at first slot of attestation slot epoch. As we will have same committee if we are on first slot of epoch or in attestation slot but in case of skip slots we would have to roll state unnecessary

Reply:
Spec says:
> Let head_state be the state of head_block processed through any empty slots up to the assigned slot using process_slots(state, slot).